### PR TITLE
CLIO-42190 thTable hover highlight fix

### PR DESCRIPTION
**Pull request by craigcarlyle**

### CODE REVIEW

File: src/lib/thTable/thTable.directive.coffee [hunk 1]
```diff
@@ -7,9 +7,31 @@ angular.module 'ThemisComponents'
       delegate: '='
     bindToController: true
     controllerAs: 'thTable'
-    controller: ($scope, $element) ->
+    controller: ($scope, $element, $timeout) ->
       thTableKeyboardNavigation $element, $scope
 
+      @mouseOver = (event) ->
+        $timeout ->
```
Comment (JeffMarvin) [inline src/lib/thTable/thTable.directive.coffee]: I am curious, what is timeout needed for here?

File: src/lib/thTable/thTable.directive.coffee [hunk 2]
```diff
@@ -7,9 +7,31 @@ angular.module 'ThemisComponents'
...
+          eventRows = getEventRows event
+          angular.element(eventRows.hoverRow).addClass 'th-table-hover-row'
```
Comment (JeffMarvin) [inline src/lib/thTable/thTable.directive.coffee]: Here and elsewhere in the PR, styleguide sez double quotes by default.

File: src/lib/thTable/thTable.directive.coffee [hunk 3]
```diff
@@ -7,9 +7,31 @@ angular.module 'ThemisComponents'
...
+          angular.element(eventRows.partnerRow).addClass 'th-table-hover-row'
+
+      @mouseLeave = (event) ->
+        $timeout ->
+          eventRows = getEventRows event
+          angular.element(eventRows.hoverRow).removeClass 'th-table-hover-row'
+          angular.element(eventRows.partnerRow).removeClass 'th-table-hover-row'
+
+      getEventRows = (event) ->
```
Comment (JeffMarvin) [inline src/lib/thTable/thTable.directive.coffee]: I see how this logic fixes the bug, but this treatment feels brittle. In the future, how much do you think we will want to generalize this to different types of rows? Any thoughts on a way to, maybe, pass a set or bind a list of rows that behave together? I say this without a full solution in mind, just thinking out loud....

File: src/lib/thTable/thTable.directive.spec.coffee [hunk 4]
```diff
@@ -81,3 +82,47 @@ describe "ThemisComponents: Directive: thTable", ->
                     .text()
                     .trim()
     expect(renderedData).toEqual outsider
+
```
Comment (JeffMarvin) [inline src/lib/thTable/thTable.directive.spec.coffee]: Woo! tests! 🎆

File: src/lib/thTable/thTable.directive.coffee (hunk 1 above)
Comment (craigcarlyle) [inline src/lib/thTable/thTable.directive.coffee]: Addressed.

File: src/lib/thTable/thTable.directive.coffee (hunk 2 above)
Comment (craigcarlyle) [inline src/lib/thTable/thTable.directive.coffee]: Addressed.

File: src/lib/thTable/thTable.directive.coffee (hunk 3 above)
Comment (JeffMarvin) [inline src/lib/thTable/thTable.directive.coffee]: Yup, makes sense. Especially sold on the fact that we are working on a new component for tables. Let's remember to make it better next time. ;)

File: src/lib/thTable/thTable.directive.spec.coffee [hunk 5]
```diff
@@ -7,9 +7,10 @@ describe "ThemisComponents: Directive: thTable", ->
   SimpleTableDelegate = TableHeader = element = null
 
   beforeEach angular.mock.module 'ThemisComponents'
-  beforeEach inject (_SimpleTableDelegate_, _TableHeader_) ->
+  beforeEach inject (_SimpleTableDelegate_, _TableHeader_, $timeout) ->
```
Comment (JeffMarvin) [inline src/lib/thTable/thTable.directive.spec.coffee]: This can go now, as well as line 13 declaring `timeout`.

**Pull request merged**